### DISCUSSION TITLE
go/pkg/mod/github.com/projectdiscovery/hmap@v0.0.11/store/hybrid/hybrid.go:10:2: //go:build comment without // +build comment

### DISCUSSION THREAD

**Discussion by kali64amd** [projectdiscovery/nuclei#3562] [Q&A]
home/shinobi/go/pkg/mod/github.com/projectdiscovery/hmap@v0.0.11/store/hybrid/hybrid.go:10:2: //go:build comment without // +build comment
home/shinobi/go/pkg/mod/github.com/projectdiscovery/interactsh@v1.1.2/pkg/options/server_options.go:5:2: //go:build comment without // +build comment
home/shinobi/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.9.1/cmd/nuclei/main.go:18:2: //go:build comment without // +build comment
home/shinobi/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.9.1/pkg/protocols/common/generators/validate.go:11:2: //go:build comment without // +build comment
home/shinobi/go/pkg/mod/github.com/projectdiscovery/utils@v0.0.18/reader/reader_keypress.go:9:2: //go:build comment without // +build comment
home/shinobi/go/pkg/mod/github.com/mattn/go-runewidth@v0.0.14/runewidth.go:7:2: //go:build comment without // +build comment

**Comment by Mzack9999** [comment]
@kali64amd without more specific info we are not able to provide help or investigate the issue. Could you resubmit the issue filling up all the sections at https://github.com/projectdiscovery/nuclei/issues/new?assignees=&labels=Type%3A+Bug&template=issue-report.md ? Thanks!

**Comment by jimen0** [comment]
@kali64amd are you using Go 1.16 by any chance? This shouldn't happen in currently supported Go versions (1.20 and 1.19).

@Mzack9999 `//go:build` directives were introduced in 1.17, until then it was `// +build` (see https://go.googlesource.com/proposal/+/master/design/draft-gobuild.md). `v0.0.14` of the `go-runewidth` uses both directive types, legacy and the current one, so this shouldn't be a problem with supported Go versions https://github.com/mattn/go-runewidth/blob/v0.0.14/runewidth_windows.go#L1-L4

It's a problem with 1.16 and earlier, though. More details on the 1.16 release notes: https://go.dev/doc/go1.16

> The new [go/build/constraint](https://go.dev/pkg/go/build/constraint/) package parses build constraint lines, both the original // +build syntax and the //go:build syntax that will be introduced in Go 1.17. This package exists so that tools built with Go 1.16 will be able to process Go 1.17 source code. See [https://golang.org/design/draft-gobuild](https://go.dev/design/draft-gobuild) for details about the build constraint syntaxes and the planned transition to the //go:build syntax. Note that //go:build lines are not supported in Go 1.16 and should not be introduced into Go programs yet.

From my experience with this type of issues it happens when people use 1.16 to compile modules that only support 1.17 onwards.

@kali64amd could you please confirm your current `go env` and `go version` output? Thanks.

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
